src/modules/oal/consumer/oal_consumer.cpp

```diff
@@ -62,15 +62,41 @@ extern "C" {
 
 namespace caspar { namespace oal {
 
+struct configuration
+{
+    std::wstring audio_device_name;
+};
+
 class device
 {
-    ALCdevice*  device_  = nullptr;
-    ALCcontext* context_ = nullptr;
+    ALCdevice*          device_  = nullptr;
+    ALCcontext*         context_ = nullptr;
+    const configuration config;
 
   public:
-    device()
+    device(const configuration& config)
```

Comment:
Since you are storing a copy of `config` inside `device`, you can pass by value and take advantage of the move semantics.